Debug-end-to-end-tests.md

```diff
@@ -0,0 +1,112 @@
+   * Look for the stack trace and error message of the _first_ error. The trace might point you to where the error was thrown in the test code, and the message may help explain what went wrong.
+
+3. If you don't understand what the error message means, search for it online. Also look through the test code and `core/test/protractor_utils/action.js` to see if the error message (or something like it) is in our test code.
+
+4. Enable [video recordings](#downloading-screen-recordings) and rerun the test until you reproduce the error. Then watch the video recordings and follow along in the test code. Try and understand why the error was thrown.
+
+5. Try and reproduce the error locally. If you succeed, you can use your [local debugger](#using-the-debugger) to investigate.
+
+## Using the debugger
+
+1. Add a break-point in the code you want the control to stop at by adding the line `debugger;`. For example:
+
+   ```js
+   ...
+   await adminPage.get();
+   await adminPage.updateRole('moderator1', 'moderator');
+   debugger;
+   await adminPage.viewRolesbyUsername('moderator1');
+   ...
+   ```
+
+2. Run the e2e script with the flag `--debug_mode`. For example,
+
+   ```console
+   python -m scripts.run_e2e_tests --debug_mode --suite="topicAndStoryEditor"
+   ```
+
+3. Wait for the script to show the following log:
+
+   ```text
+   Debugger listening on ws://127.0.0.1:9229/e4779cc6-72e9-4d8d-889e-1fb3b2628781
+   For help, see: https://nodejs.org/en/docs/inspector
+   ```
+
+4. At this point, go to `chrome://inspect/#devices` on your Chrome browser.
+
+5. Click on "inspect" under Remote Target (see screenshot below).
+
+   ![Inspect Page](https://user-images.githubusercontent.com/11008603/88563290-714bac80-d04f-11ea-8b36-fc43c66d6e3d.png)
+
+6. A Chrome dev tools instance will open up and the e2e test should start executing in a new window.
+
+7. The control will stop at the point where the debugger statement was added. You can now choose to inspect elements, log variables in the test, or add more break-points.
+
+## Rerunning with SSH
+
+CircleCI allows debugging using SSH. For details, please read [this](https://circleci.com/docs/2.0/ssh-access-jobs/#steps). Debugging with SSH only reruns that particular job, so it is a great way to rerun a passing test to see if it flakes without rerunning all the tests.
+
+## Downloading screenshots
+
+We capture screenshots of failing tests. On CircleCI, these are available under the `Artifacts` tab of the failure log page. You may also want to reference the [CircleCI artifacts documentation](https://circleci.com/docs/2.0/artifacts/). On GitHub Actions, look for an `Artifacts` link in the upper right where you can download a zip file of the screenshots.
+
+Here's an example of what artifacts on CircleCI look like:
+
+![CircleCI artifacts page with hyperlinks to screenshots and reports](https://user-images.githubusercontent.com/19878639/111242142-f1ba2000-85d4-11eb-8bf1-66cfbbf71975.png)
```

Comment:
Should we update this with Github-Action's artifacts tab?

Reply:
done